rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -1331,7 +1330,25 @@ impl NewCanisterQueues {
             guaranteed_response_memory_reservations,
             input_queues_reserved_slots,
             output_queues_reserved_slots,
-            transient_stream_responses_size_bytes: 0,
+            transient_stream_guaranteed_responses_size_bytes,
+        }
+    }
+}
+
+/// Returns a function that determines the input queue type (local or remote) of
+/// a given sender, based on a the set of all local canisters, plus
+/// `own_canister_id` (since Rust's ownership rules would prevent us from
+/// mutating a canister's queues if they were still under `local_canisters`).
+#[allow(dead_code)]
+fn input_queue_type_fn<'a>(
+    own_canister_id: &'a CanisterId,
+    local_canisters: &'a BTreeMap<CanisterId, CanisterState>,
+) -> impl Fn(&CanisterId) -> InputQueueType + 'a {
+    move |sender| {
```

Comment:
I'm not sure where the `sender` comes from?

Reply:
As per the doc comment, this returns a function that, given a sender, will determine whether it's a local or remote sender. Unfortunately, the enum it produces is named `InputQueueType` rather than something like `SenderGroup` or `SenderProximity` or similar.

Also, `SenderGroup` and `SenderProximity` and all other variations I could come up with were not any better than `InputQueueType`, so I didn't rename it.

Any decent naming suggestions are welcome, though.